lib/Conversion/ImportVerilog/Statements.cpp

```diff
@@ -463,6 +473,31 @@ struct StmtVisitor {
     return success();
   }
 
+  // Handle immediate assertion statements.
+  LogicalResult visit(const slang::ast::ImmediateAssertionStatement &stmt) {
+    auto cond = context.convertRvalueExpression(stmt.cond);
+    cond = context.convertToBool(cond);
+    if (!cond)
+      return failure();
+
+    auto defer = convertDeferAssert(stmt.isDeferred, stmt.isFinal);
+    switch (stmt.assertionKind) {
+    case slang::ast::AssertionKind::Assert:
+      builder.create<moore::AssertOp>(loc, defer, cond, StringAttr{});
+      return success();
+    case slang::ast::AssertionKind::Assume:
+      builder.create<moore::AssumeOp>(loc, defer, cond, StringAttr{});
+      return success();
+    case slang::ast::AssertionKind::CoverProperty:
+      builder.create<moore::CoverOp>(loc, defer, cond, StringAttr{});
+      return success();
+    default:
+      break;
+    }
+    mlir::emitError(loc, "all immediate assertions handled");
```

Comment:
Can you make this complain about unsupported assertions kinds? I assume there are more than just the three listed above. You could print something like:
```suggestion
    mlir::emitError(loc) << "unsupported immediate assertion kind: " << slang::ast::toString(stmt.assertionKind);
```

Reply:
This new error message is obviously better.